libraries/lib-wave-track/WaveClip.cpp

```diff
@@ -150,6 +150,16 @@ AudioSegmentSampleView WaveClip::GetSampleView(
       start + TimeToSamples(mTrimLeft), length, mayThrow);
 }
 
+AudioSegmentSampleView WaveClip::GetSampleView(
+   size_t iChannel, double t0, double t1, bool mayThrow) const
+{
+   assert(iChannel < GetWidth());
+   const auto start = TimeToSamples(std::max(0., t0));
+   const auto length =
+      (std::min(GetNumSamples(), TimeToSamples(t1)) - start).as_size_t();
```

Comment:
What's your proof that the assertions in `as_size_t` are satisfied -- that there is no narrowing of a negative or a too-large `SampleCount` value?

Reply:
My own, old comment in the file says
`   // Expect that the difference of t1 and t0 is the part of a track played
    // in about 1/20 second (ticks of TrackPanel timer), so this won't overflow
`
I didn't make a really rigorous proof then.  But can we do better?

Stretches aren't limited yet to "reaonable" bounds and also the calculation above is specific to default sample rate.